ledgers/ccf/scripts/start_ccf_network.sh

```diff
@@ -16,45 +16,101 @@
 
 # -----------------------------------------------------------------
 # -----------------------------------------------------------------
-F_SERVICEHOME="$( cd -P "$( dirname ${BASH_SOURCE[0]} )/.." && pwd )"
-source ${F_SERVICEHOME}/bin/lib/pdo_common.sh
-
+source ${PDO_HOME}/bin/lib/common.sh
 check_python_version
 
-if [ -f ${F_SERVICEHOME}/run/cchost.pid ]; then
-    if ps -p $(cat ${F_SERVICEHOME}/run/cchost.pid) > /dev/null
+# -----------------------------------------------------------------
+# Process command line arguments
+# -----------------------------------------------------------------
+F_INTERFACE=${PDO_HOSTNAME:-${HOSTNAME}}
+F_PORT=6600
+
+SCRIPT_NAME=$(basename ${BASH_SOURCE[-1]} )
+USAGE='-i|--interface [hostname] -p|--port [port]'
+SHORT_OPTS='i:p:'
+LONG_OPTS='interface:,port:'
+
+TEMP=$(getopt -o ${SHORT_OPTS} --long ${LONG_OPTS} -n "${SCRIPT_NAME}" -- "$@")
+if [ $? != 0 ] ; then echo "Usage: ${SCRIPT_NAME} ${USAGE}" >&2 ; exit 1 ; fi
+
+eval set -- "$TEMP"
+while true ; do
+    case "$1" in
+        -i|--interface) F_INTERFACE="$2" ; shift 2 ;;
+        -p|--port) F_PORT="$2" ; shift 2 ;;
+        --help) echo "Usage: ${SCRIPT_NAME} ${USAGE}"; exit 0 ;;
+    	--) shift ; break ;;
+    	*) echo "Internal error!" ; exit 1 ;;
+    esac
+done
+
+# the CCF sandbox script creates the python virtual environment
+# in the directory from which it is executed so we need to make
+# sure we run it in the install directory... this is generally bad
+# practice and we should probably move away from sandbox.
+cd ${PDO_HOME}/ccf
+
```

Comment:
good point. but the earlier check for a running ccf SHOULD have caught that case. the problem is once the pid file is removed, there is no easy way to check this.

Reply:
Because it was an old instance of ccf, there was no pid for that.

Reply:
Actually... the problem is that line new-70 starts the sandbox in background and assumes no failure -- in my case, it failed because the port was already taken.

A more robust check is to extend the while condition to wait for the keys if the process is alive and the keys are not available -- the liveness check should be the same as in line new-54.

Reply:
> Because it was an old instance of ccf, there was no saved pid for that.

understood, but we cannot, nor should we try, to catch every single failure condition in startup. i'm adding the lsof check for a process on the same port. that should provide a high level check, but it certainly will not catch all conditions. feel free to submit PRs for additional checks if you would like to add them.